pandas/tests/scalar/timedelta/test_timedelta.py

```diff
@@ -354,6 +354,13 @@ def test_unit_parser(self, units, np_unit, wrapper):
             result = Timedelta('2{}'.format(unit))
             assert result == expected
 
+    @pytest.mark.parametrize('units', ['Y', 'y', 'M'])
+    def test_unit_M_Y_deprecated(self, units):
+        with tm.assert_produces_warning(FutureWarning):
```

Comment:
these need separate tests for catching the FutureWarning; What you did didn't catch for Timedelta

Reply:
only test the scalar here, the others are tested elsewhere